### PR TITLE
Wrap request to avoid modifying attributes of the original request

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10379
Or spring mvc instrumentation tries to update the root span name/set route at the end of the request if it wasn't already set earlier. This usually happens when the request is rejected for some reason and request processing does not reach the part where request handler is determined and where we usually do the name update. If that is the case we try to find the handler that matches current request and use the patter it matches as the route. This process alters request attributes which is not desired. This PR wraps the original request so that its attributes could be preserved.

### CODE REVIEW

**Review by kenfinnigan — APPROVED** [review]
Thanks for the quick turnaround @laurit!

**Review by trask — APPROVED** [review]
nice

File: instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/main/java/org/springframework/web/servlet/v3_1/OpenTelemetryHandlerMappingFilter.java [hunk 1]
```diff
@@ -26,41 +28,32 @@
...
-            }
-          }
+        if (findMapping(request)) {
+          // Name the parent span based on the matching pattern
+          // Let the parent span resource name be set with the attribute set in findMapping.
+          return SpringWebMvcServerSpanNaming.SERVER_SPAN_NAME.get(context, request);
         }
+        // we are calling this method with a HttpServletRequestWrapper that does not delegate
+        // setting attributes to the original request, so we can skip restoring the attributes to
+        // their original values
```
[the remark this answers is not in the document]
Comment (kenfinnigan) [reply instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/main/java/org/springframework/web/servlet/v3_1/OpenTelemetryHandlerMappingFilter.java:55]: The previous version of the code was capturing the existing value of `PATH_ATTRIBUTE` and resetting it back on the request after `findMapping()` had executed.

With the new code, there are no longer any sideeffects which makes the restoration unnecessary.

Maybe the comment isn't needed without the previous code being there anymore?

[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/spring/spring-webmvc/spring-webmvc-3.1/javaagent/src/main/java/org/springframework/web/servlet/v3_1/OpenTelemetryHandlerMappingFilter.java]: I removed the comment. The intent was to explain why we don't attempt to restore the request to the original state but I guess it wasn't obvious that the request is modified here in the first place. There is already a comment where `HttpServletRequestWrapper` is created that covers this.

**Pull request merged**